core/subsonic-api/src/main/kotlin/org/moire/ultrasonic/api/subsonic/SubsonicAPIClient.kt

```diff
@@ -73,7 +74,19 @@ class SubsonicAPIClient(
                 .addQueryParameter("c", config.clientID)
                 .addQueryParameter("f", "json")
                 .build()
-            chain.proceed(originalRequest.newBuilder().url(newUrl).build())
+            var newRequestBuilder = originalRequest.newBuilder().url(newUrl)
+            if (originalRequest.url.username.isNotEmpty() &&
+                originalRequest.url.password.isNotEmpty()
+            ) {
+                newRequestBuilder = newRequestBuilder.addHeader(
+                    "Authorization",
+                    Credentials.basic(
+                        originalRequest.url.username,
+                        originalRequest.url.password
+                    )
+                )
```

Comment:
Why instantiate two Builders after each other?
Just remove the build() call from the first builder, reuse it in your logic,
and then call build() in the end.

Reply:
You are totally right, and quickly reading I didn't see the difference :)
But the reassignment on Line 81 should not be needed. The reason that these methods on the builder returns a Builder ist to enable chaining. The instance of the builder that you call the method on is the same instance that is being returned.